### ISSUE TITLE
OneStatementPerLine is affected by annotation

### ISSUE THREAD

**Issue by sarser2048** [checkstyle/checkstyle#12769]
I have read check documentation: https://checkstyle.sourceforge.io/config_coding.html#OneStatementPerLine
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Main.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="severity" value="error"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="OneStatementPerLine"/>
  </module>
</module>

/var/tmp $ cat Main.java
public class Main {
  String str = "123";
  ;  // report a OneStatementPerLine warning here
  @MyAnnotation
  String str = "123";
  ;  // no OneStatementPerLine Warning
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.3.3-all.jar -c config.xml YOUR_FILE.java
Starting audit...
[ERROR] /path/tofile:3:3: Only one statement per line allowed. [OneStatementPerLine]
Audit done.
Checkstyle ends with 1 errors.
```
---

**Describe what you expect in detail.**
---

UPDATED:
OneStatementPerLine  should not do any violation on such code . 

https://checkstyle.sourceforge.io/checks/coding/emptystatement.html#EmptyStatement should care about empty `;` on a line.

**Comment by nrmancuso** [comment]
@sarser2048 please follow issue template

**Comment by sarser2048** [comment]
@nrmancuso Hi, I have changed my issue based on the template, thanks.

**Comment by romani** [comment]
Please update line numbers they are mismatching, hard to understand what is current behavior.
Please use trailing comment on line where is violation or you expect violation.

**Comment by sarser2048** [comment]
Hi @romani , I have updated my description in the end of issue.

**Comment by romani** [comment]
I code you have line 3 there violation, but CLI output only shows at line 5. CLI shows errors but you in comment think it 😛 s warning.

I really confused. Please clarify.

**Comment by romani** [comment]
https://github.com/checkstyle/checkstyle/actions/runs/12615966172/job/35156649766#step:6:49

**Comment by mohitsatr** [comment]
im on it.

**Comment by nrmancuso** [comment]
@romani, the last two edits have made this issue difficult to understand, and I’m unclear on the expectations here. Initially, @sarser2048 reported one output message, but it was later changed.  

After reviewing the check documentation, I believe nothing in the provided example should trigger a violation. Can you confirm if this is the intended behavior?

**Comment by mohitsatr** [comment]
if the same code is present in a method, check does not show any violation.

with same config:
```java
public class Test {
    String str = "123";
    ;  // violation 
    @SuppressWarnings("unused")
    String str1 = "123";
    ;  
    String str2 = "123";
    ; // violation

    public void fooDummy() {
        String str = "123";
        ; // false negative 
        String str2 = "123";
        ; // false negative 
        @SuppressWarnings("")
        String str3 = "122";;
        ;
    }
}

$ java -jar testing/checkstyle-10.21.0-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /mnt/5D92528E6B945467/test/Test.java:3:5: Only one statement per line allowed. [OneStatementPerLine]
[ERROR] /mnt/5D92528E6B945467/test/Test.java:8:5: Only one statement per line allowed. [OneStatementPerLine]
Audit done.
Checkstyle ends with 2 errors.
```


there is slight difference in AST 
```
      |--VARIABLE_DEF -> VARIABLE_DEF [7:4]
        |   |--MODIFIERS -> MODIFIERS [7:4]
        |   |--TYPE -> TYPE [7:4]
        |   |   `--IDENT -> String [7:4]
        |   |--IDENT -> str2 [7:11]
        |   |--ASSIGN -> = [7:16]
        |   |   `--EXPR -> EXPR [7:18]
        |   |       `--STRING_LITERAL -> "123" [7:18]
        |   `--SEMI -> ; [7:23]
        |--SEMI -> ; [8:4]

for fooDummy method : 
      |--SLIST -> { [10:27]
        |       |--VARIABLE_DEF -> VARIABLE_DEF [11:8]
        |       |   |--MODIFIERS -> MODIFIERS [11:8]
        |       |   |--TYPE -> TYPE [11:8]
        |       |   |   `--IDENT -> String [11:8]
        |       |   |--IDENT -> str [11:15]
        |       |   `--ASSIGN -> = [11:19]
        |       |       `--EXPR -> EXPR [11:21]
        |       |           `--STRING_LITERAL -> "123" [11:21]
        |       |--SEMI -> ; [11:26]
        |       |--**EMPTY_STAT** -> ; [12:8]

```

**Comment by rnveach** [comment]
This issue seems like this is trying to encroach on `EmptyStatement`.

**Comment by Atharv3221** [comment]
> if the same code is present in a method, check does not show any violation.
> 
> with same config:
> 
> public class Test {
>     String str = "123";
>     ;  // violation 
>     @SuppressWarnings("unused")
>     String str1 = "123";
>     ;  
>     String str2 = "123";
>     ; // violation
> 
>     public void fooDummy() {
>         String str = "123";
>         ; // false negative 
>         String str2 = "123";
>         ; // false negative 
>         @SuppressWarnings("")
>         String str3 = "122";;
>         ;
>     }
> }
> 
> $ java -jar testing/checkstyle-10.21.0-all.jar -c config.xml Test.java 
> Starting audit...
> [ERROR] /mnt/5D92528E6B945467/test/Test.java:3:5: Only one statement per line allowed. [OneStatementPerLine]
> [ERROR] /mnt/5D92528E6B945467/test/Test.java:8:5: Only one statement per line allowed. [OneStatementPerLine]
> Audit done.
> Checkstyle ends with 2 errors.
> there is slight difference in AST
> 
> ```
>       |--VARIABLE_DEF -> VARIABLE_DEF [7:4]
>         |   |--MODIFIERS -> MODIFIERS [7:4]
>         |   |--TYPE -> TYPE [7:4]
>         |   |   `--IDENT -> String [7:4]
>         |   |--IDENT -> str2 [7:11]
>         |   |--ASSIGN -> = [7:16]
>         |   |   `--EXPR -> EXPR [7:18]
>         |   |       `--STRING_LITERAL -> "123" [7:18]
>         |   `--SEMI -> ; [7:23]
>         |--SEMI -> ; [8:4]
> 
> for fooDummy method : 
>       |--SLIST -> { [10:27]
>         |       |--VARIABLE_DEF -> VARIABLE_DEF [11:8]
>         |       |   |--MODIFIERS -> MODIFIERS [11:8]
>         |       |   |--TYPE -> TYPE [11:8]
>         |       |   |   `--IDENT -> String [11:8]
>         |       |   |--IDENT -> str [11:15]
>         |       |   `--ASSIGN -> = [11:19]
>         |       |       `--EXPR -> EXPR [11:21]
>         |       |           `--STRING_LITERAL -> "123" [11:21]
>         |       |--SEMI -> ; [11:26]
>         |       |--**EMPTY_STAT** -> ; [12:8]
> ```

violation was expected for line 6
>     @SuppressWarnings("unused")
>     String str1 = "123";
>     ;

**Comment by romani** [comment]
@Atharv3221 , 
https://checkstyle.sourceforge.io/checks/coding/emptystatement.html#EmptyStatement
this Check should care about empty `;` on a line.

OneStatementPerLine should be updated to not do violation on what author is showed in issue description.
Lets update OneStatementPerLine to not do violations on such code.

**Comment by Atharv3221** [comment]
> @Atharv3221 , 
> https://checkstyle.sourceforge.io/checks/coding/emptystatement.html#EmptyStatement
> this Check should care about empty `;` on a line.
> 
> OneStatementPerLine should be updated to not do violation on what author is showed in issue description.
> Lets update OneStatementPerLine to not do violations on such code.

Yes it's fixed in https://github.com/checkstyle/checkstyle/pull/18567

**Comment by Atharv3221** [comment]
here its false positive
